replication-api-impl/src/main/java/com/connexta/ion/replication/api/impl/persistence/ReplicationItemManagerImpl.java

```diff
@@ -80,17 +93,46 @@ public void deleteItem(String metadataId, String source, String destination) {
   }
 
   @Override
-  public List<String> getFailureList(int maximumFailureCount, String source, String destination) {
-    return itemRepository
-        .findByFailureCountBetweenAndSourceAndDestination(
-            1,
-            maximumFailureCount - 1,
-            source,
-            destination,
-            PageRequest.of(DEFAULT_START_INDEX, 50))
-        .stream()
-        .map(ReplicationItemImpl::getId)
-        .collect(Collectors.toList());
+  public List<String> getFailureList(String configId) {
```

Comment:
Group queries are not supported by the Repository interface, so I had to do this manually. Eventually this query will not be needed when queues are supported.